### PR TITLE
Remove dependency on Zalando Problem library

### PR DESCRIPTION
**Pull request by lwitkowski**
Main changes: 
`Problem` interface from Zalando Lib is no longer a core class used everywhere in the extension - it is replaced by `HttpProblem`, which also carries http headers, and have nice 'clone builder' that helps building `HttpProblem`s based on other `HttpProblem`s.
`Status` from Zalando's lib got replaced with `Response.Status` from JaxRS everywhere as well.

Thanks to `HttpProblem` having place for http headers it was super easy to simplify `WebApplicationExceptionMapper` so that it extends `ExceptionMapperBase` again, but without extending api surface of this base class.

`ProblemUtils` class is gone.

Zalando's Problem is supported though (we provide a mapper if library is in the classpath).

This implementation is 4-5 times faster than the one based on Zalando's Problem :)

### CODE REVIEW

File: integration-test/src/test/java/com/tietoevry/quarkus/resteasy/problem/JaxRsMappersIT.java [hunk 1]
```diff
@@ -50,23 +48,11 @@ void webApplicationExceptionShouldReturnHeaders() {
                 .log().all()
                 .statusCode(status)
                 .header(HttpHeaders.RETRY_AFTER, equalTo("120"))
-                .body("title", equalTo(Status.valueOf(status).getReasonPhrase()))
+                .body("title", equalTo(Response.Status.fromStatusCode(status).getReasonPhrase()))
                 .body("status", equalTo(status))
                 .body("stacktrace", nullValue());
     }
 
-    @Test
```
Comment (pazkooda) [inline integration-test/src/test/java/com/tietoevry/quarkus/resteasy/problem/JaxRsMappersIT.java:58]: Why it is gone?

File: runtime/pom.xml [hunk 2]
```diff
@@ -40,12 +46,7 @@
             <groupId>org.zalando</groupId>
             <artifactId>problem</artifactId>
             <version>${zalando-problem.version}</version>
-        </dependency>
-        <!-- fix missing javax.annotation.concurrent.Immutable -->
-        <dependency>
-            <groupId>com.google.code.findbugs</groupId>
-            <artifactId>jsr305</artifactId>
-            <version>3.0.2</version>
+            <optional>true</optional>
```
Comment (pazkooda) [inline runtime/pom.xml:49]: 👍

File: runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/javax/ConstraintViolationExceptionMapper.java [hunk 3]
```diff
@@ -16,8 +21,36 @@
...
+                .build();
+    }
+
+    private Violation toViolation(ConstraintViolation<?> constraintViolation) {
+        return new Violation(
+                constraintViolation.getMessage(),
+                dropFirstTwoPathElements(constraintViolation.getPropertyPath()));
+    }
+
+    private String dropFirstTwoPathElements(Path propertyPath) {
```
Comment (pazkooda) [inline runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/javax/ConstraintViolationExceptionMapper.java:43]: Good time to rename it to purpose of it. My suggestion:

```java
dropMethodNameAndArgumentPostionFromPath(...)
```

**Review by pazkooda — APPROVED** [review]
LGTM, see details ;)

File: integration-test/src/test/java/com/tietoevry/quarkus/resteasy/problem/JaxRsMappersIT.java (hunk 1 above)
Comment (lwitkowski) [reply integration-test/src/test/java/com/tietoevry/quarkus/resteasy/problem/JaxRsMappersIT.java:58]: It was moved to separate ZalandoProblemMapperIT to make zalando stuff even more separated from everything.

**Pull request merged**